c7n/resources/ecs.py

```diff
@@ -525,3 +538,145 @@ def process_instance(self, client, cluster, instance):
         except (client.exceptions.NoUpdateAvailableException,
                 client.exceptions.UpdateInProgressException):
             return
+
+
+@ECSCluster.action_registry.register('tag')
+@TaskDefinition.action_registry.register('tag')
+@Service.action_registry.register('tag')
+@Task.action_registry.register('tag')
+@ContainerInstance.action_registry.register('tag')
+class TagEcsResource(Tag):
+    """Action to create tag(s) on an ECS resource
+    (ecs, ecs-task-definition, ecs-service, ecs-task, ecs-container-instance)
+
+    Requires arns in new format for tasks, services, and container-instances.
+    https://docs.aws.amazon.com/AmazonECS/latest/userguide/ecs-resource-ids.html
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: tag-ecs-service
+                resource: ecs-service
+                filters:
+                  - "tag:target-tag": absent
+                  - type: taggable
+                actions:
+                  - type: tag
+                    key: target-tag
+                    value: target-value
+    """
+    permissions = ('ecs:TagResource',)
+
+    def process_resource_set(self, client, resources, tags):
+        mid = self.manager.resource_type.id
+        for r in resources:
+            if not ecs_taggable(self.manager.resource_type, r):
+                raise PolicyExecutionError("New format ecs arn required: " + r[mid])
+
+        tags = [{'key': t['Key'], 'value': t['Value']} for t in tags]
+        for r in resources:
+            client.tag_resource(resourceArn=r[mid], tags=tags)
+
+
+@ECSCluster.action_registry.register('remove-tag')
+@TaskDefinition.action_registry.register('remove-tag')
+@Service.action_registry.register('remove-tag')
+@Task.action_registry.register('remove-tag')
+@ContainerInstance.action_registry.register('remove-tag')
+class RemoveTagEcsResource(RemoveTag):
+    """Remove tag(s) from ECS resources
+    (ecs, ecs-task-definition, ecs-service, ecs-task, ecs-container-instance)
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: ecs-cluster-remove-tag
+                resource: ecs
+                filters:
+                  - "tag:BadTag": present
+                actions:
+                  - type: remove-tag
+                    tags: ["BadTag"]
+    """
+    permissions = ('ecs:UntagResource',)
+
+    def process_resource_set(self, client, resources, keys):
+        for r in resources:
```

Comment:
if you're going to do a is taggable check in tag, it should also be present in remove-tag